confgenerator/feature.go

```diff
@@ -0,0 +1,176 @@
+package confgenerator
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+var ErrTrackingInlineStruct = errors.New("cannot have tracking on inline struct")
+
+type Feature struct {
+	Module, Kind, Type, Key, Value string
+}
+
+func ExtractFeatures(uc *UnifiedConfig) ([]Feature, error) {
```

Comment:
If you're already doing the work to generically use reflection, why not use reflection for `ExtractFeatures` itself? You can just walk the entire config struct and expose any matching field generically with a key like

`Logging.Receivers.2[apache_access].WildcardRefreshInterval`

(i.e. use the full struct path as the key)

Then this function goes away, and you don't need any names in the struct tags. Since this is only for our own internal use, it's not really a problem if the names are less friendly.

(I would also suggest identifying fields to export based on the type of the field, i.e. int and bool fields are always (?) not PII, so don't need any tags.)

Reply:
Discussed this in the design doc under the "Flatten the <key, value> pairs" section. 

Pros:
* Fewer labels.
Cons:
* More complex to query, filter by and aggregate.
* Needs heavy regex matching.

The cons outweighs the pros in this case, so keeping the `module` and `feature` is more convenient. Added a note for "alternative considered" in the design doc for the records.

Reply:
I'm not sure I follow. The querying and filtering is going to happen on the Concord table, right? So why do the metric fields matter? We can have `module` and `feature` columns in the Concord table but keep the OA-side implementation simple.

Reply:
There are some internal-only details. Let's move to the internal chats.

Reply:
Summarized the offline chat in https://github.com/GoogleCloudPlatform/ops-agent/pull/853#discussion_r986287724.